src/workerd/api/memory-cache.h

```diff
@@ -428,30 +428,16 @@ class MemoryCache: public jsg::Object {
   SharedMemoryCache::Use cacheUse;
 };
 
-// Data structure that maps unique cache identifiers to cache instances.
-// This allows separate isolates to access the same in-memory caches.
-class MemoryCacheMap {
+// The MemoryCacheProvider provides the internal implementation of the MemoryCache mechanism.
```

Comment:
What's the benefit of virtualizing `MemoryCacheMap` into `MemoryCacheProvider`? All this type implements is the mapping of cache IDs to `SharedMemoryCache` objects. If we introduced a new memcache-backed cache, this lookup would presumably still work the same? It's the `SharedMemoryCache` object itself which needs to be extensible. And even there, I don't think it would be accomplished by virtualization, but rather extending it to accept some sort of fallback cache storage that it checks before giving up and calling the callback.

Reply:
I'd say, let's not make any movement in that direction until we know where we're going. Keep `MemoryCacheMap` as a concrete class for now.

Reply:
Aside from this... I think we need to determine if the way limits are specified/suggested for the shared class via the binding configuration continues to make sense with the memcache-based implementation. If not, we might want to refactor that a bit here also.